packet/accumulator.go

```diff
@@ -30,46 +30,38 @@ import (
 	"github.com/Comcast/gots"
 )
 
-var (
-	emptyByteArray []byte
-)
-
-type doneFunc func([]byte) (bool, error)
```

Comment:
Why eliminate this type?

Reply:
It gives you very little. The only thing it saves you from is another named function type, but this is unexported to the surface is minuscule and it causes readability and documentation to suffer.

To expand on this, your `NewAccumulator` takes a `doneFunc`, but what is this? A users needs to read the source because it's unexported and so doesn't show up in the godoc. And because it's unexported they have to use a `func([]byte) (bool, error)` anyway which is assignable to a `doneFunc` but is not the same type. If it were exported there would be a marginal benefit, but still it would impact on readability since "What is a packet.DoneFunc anyway?.... [navigates to definition... Ah, OK." and still most people will use an unnamed type closure or function value anyway.

Reply:
I think there is some value in the label `doneFunc` as it makes it easier to understand what is happening. That label can be used in `accumulator` and `NewAccumulator` instead.

Reply:
I have no plan for any additional implementations. IIRC, the possibility existed when we created it, but obviously that has not come to be. It's possible that `doneFunc` makes sense to me because I am used to starting at it. I am a big fan of self documenting code and I think that `func` types are a great way to do that.

Reply:
WRT `doneFunc`, it actually reduced code documentation except for people reading the source, and moves the identity of the type away from the use, to the definition of the type. To demonstrate, read the godoc for [`packet.NewAccumulator`](https://godoc.org/github.com/Comcast/gots/packet#NewAccumulator). Now, as a person who wants to use `packet.NewAccumulator`, but has not read the source and is new to gots (there are many Go users who don't read source of packages they use - I don't understand why, but it's a thing), exactly what is a `doneFunc` type? The documentation is completely absent.because the type not exported, so you would need to navigate to the `package.NewAccumulator` definition in the source in the hope that `type doneFunc` is defined *near* near it, which in this case it is-ish. If you look at the case in the stdlib that take func types, generally the types are only named types when they have methods on them, or they are usually not user-facing types [`http.HandlerFunc`](https://golang.org/pkg/net/http/#HandlerFunc) is an obvious case of user-facing named func type, but it has a [method](https://golang.org/pkg/net/http/#HandlerFunc.ServeHTTP).

For the use of interfaces here, I wouldn't. Again this use reduced the visibility of documentation because you lose links from the godoc to method implementations. You can see this by navigating to [`packet.Accumulator`](https://godoc.org/github.com/Comcast/gots/packet#Accumulator). If it were a concrete type, you would be able to follow the links to source. This doesn't help those mentioned above who don't read the source, but is a been for people who do when the documentation is not helpful enough.

Reply:
BTW There is no handling of the case of a nil parameter here. Is there a sensible default? If not, it should probably be guarded and `NewAccumulator` should return an error.

Also, I'm wondering if it's not more sensible to just have the `packets` and `f`fields exported. Is there anything you are particularly protecting against there?